sway/input/cursor.c

```diff
@@ -743,11 +746,11 @@ static void handle_tool_button(struct wl_listener *listener, void *data) {
 		(enum zwp_tablet_pad_v2_button_state)event->state);
 }
 
-static void check_constraint_region(struct sway_cursor *cursor) {
+static void check_constraint_region(struct sway_cursor *cursor, bool do_warp) {
```

Comment:
Note for reviewer: this flag is a bit of an awkward way to do this, in theory the entire warp logic could be moved to line 1285 (which is the only caller with `do_wrap = true`), but I opted for this way to minimize future churn, since the call on line 1285 is a workaround for another issue.